pkg/reconciler/pipelinerun/resources/pipelinerunresolution_test.go

```diff
@@ -2660,3 +2701,79 @@ func TestPipelineRunState_GetFinalTasks(t *testing.T) {
 		})
 	}
 }
+
+func TestResolvePipeline_WhenExpressions(t *testing.T) {
+	names.TestingSeed()
+	tName1 := "pipelinerun-mytask1-9l9zj-always-true-mz4c7"
+	tName2 := "pipelinerun-mytask1-9l9zj-always-true-mssqb"
+
+	t1 := &v1beta1.TaskRun{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: tName1,
+		},
+	}
+
+	t2 := &v1beta1.TaskRun{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: tName2,
+		},
+	}
+
+	ptwe1 := v1beta1.WhenExpression{
+		Input:    "foo",
+		Operator: selection.In,
+		Values:   []string{"foo"},
+	}
+
+	ptwe2 := v1beta1.WhenExpression{
+		Input:    "foo",
+		Operator: selection.NotIn,
+		Values:   []string{"bar"},
+	}
+
+	pts := []v1beta1.PipelineTask{{
+		Name:            "mytask1",
+		TaskRef:         &v1beta1.TaskRef{Name: "task"},
+		WhenExpressions: []v1beta1.WhenExpression{ptwe1, ptwe2},
+	}}
+
+	providedResources := map[string]*resourcev1alpha1.PipelineResource{}
+
+	getTask := func(name string) (v1beta1.TaskInterface, error) { return task, nil }
+	getClusterTask := func(name string) (v1beta1.TaskInterface, error) { return nil, errors.New("should not get called") }
+	getCondition := func(name string) (*v1alpha1.Condition, error) { return &condition, nil }
+	pr := v1beta1.PipelineRun{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "pipelinerun",
+		},
+	}
+
+	tcs := []struct {
+		name       string
+		getTaskRun resources.GetTaskRun
+	}{
+		{
+			name: "When Expressions exist",
+			getTaskRun: func(name string) (*v1beta1.TaskRun, error) {
+				switch name {
+				case "pipelinerun-mytask1-9l9zj-always-true-0-mz4c7":
+					return t1, nil
+				case "pipelinerun-mytask1-9l9zj":
+					return &trs[0], nil
+				case "pipelinerun-mytask1-9l9zj-always-true-1-mssqb":
+					return t2, nil
+				}
+				return nil, fmt.Errorf("getTaskRun called with unexpected name %s", name)
+			},
+		},
+	}
+
+	for _, tc := range tcs {
```

Comment:
i think there is only one test case here? i usually avoid putting in table driven tests with just one case unless im SURE we want more than one